tiff-2-png-converter.py

```diff
@@ -8,189 +9,273 @@
 import tifffile  as tiff
 import numpngw
 from progress_bar import *
+from image_process_helper import *
+from scipy.interpolate import interp1d
+import re
+
+
 
-verbose = False
-
-
-
-
-def convert_tiff_directory_action(source_path, dest_path):
-    list_files,width, heigth = list_folder_files(source_path)
-    num_images = len(list_files)
-    image_r = [None] * num_images
-    image_g = [None] * num_images
-    image_b = [None] * num_images
-    images = [None] * num_images
-    bits_per_sample = ""
-    printProgressBar(0, num_images, prefix = 'grouping channels:', suffix = 'Complete', length = 50)
-    for i in range(num_images):
-        file_full_path = list_files[i]
-        img = tiff.imread(list_files[i])
-        if bits_per_sample=="":
-            bits_per_sample = img.dtype
-        elif bits_per_sample!=img.dtype:
-            print("image " + file_full_path + "has a different sample type "+bits_per_sample+" expected: "+bits_per_sample)
-            sys.exit("Check images bits per sample tag")
-        name, extension = os.path.splitext(file_full_path)
-        if "ch1" in name:
-             if verbose:
-                print("Load Image " + str(i) + " Channel 1 - "+ file_full_path )
-             image_r[i] = img
-        elif "ch2" in name:
-             if verbose:
-                print("Load Image " + str(i) + " Channel 2 - "+ file_full_path )
-             image_g[i] = img
-        elif "ch3" in name:
-             if verbose:
-                print("Load Image " + str(i) + " Channel 3 - "+ file_full_path )
-             image_b[i] = img
+class Tiff_Processor():
```

Comment:
Class names recommendation is ` CapWords` convention

Reply:
Renamed it following CapWords  convention